packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -294,11 +317,30 @@ class AOTSnapshotter {
       '-o', appLib,
       assemblyO,
     ];
-    final RunResult linkResult = await _xcode.clang(linkArgs);
-    if (linkResult.exitCode != 0) {
-      _logger.printError('Failed to link AOT snapshot. Linker terminated with exit code ${compileResult.exitCode}');
+
+    result = await _xcode.clang(linkArgs);
+    if (result.exitCode != 0) {
+      _logger.printError('Failed to link AOT snapshot. Linker terminated with exit code ${result.exitCode}');
     }
-    return linkResult;
+
+    if (extractAppleDebugSymbols) {
+      result = await _xcode.dsymutil(<String>['-o', '$frameworkDir.dSYM', appLib]);
+      if (result.exitCode != 0) {
+        _logger.printError('Failed to generate dSYM - dsymutil terminated with exit code ${result.exitCode}');
+      }
+
+      if (stripAfterBuild) {
+        // See https://www.unix.com/man-page/osx/1/strip/ for arguments
+        result = await _xcode.strip(<String>['-S', appLib, '-o', appLib]);
```

Comment:
I've originally added `strip -S -x` here, to also strip all local symbols. Turns out, the current behaviour on the master branch is not to do that and instead keeps those symbols:

```shell-script
% nm -a ./build-master/ios/Release-iphoneos/App.framework/App | wc 
   15850   47544 1011424
% nm -a ./build-strip-s/ios/Release-iphoneos/App.framework/App | wc  
   15850   47544 1011424
% nm -a ./build-strip-s-x/ios/Release-iphoneos/App.framework/App | wc 
       7      15     232
   ```

Is anyone aware why we need those symbols? Might be a good followup issue, as it reduces the framework size, for the sample I've used from about 7 MB to 6MB.

Reply:
@vaind AFAIK only the 4 symbols mentioned in https://github.com/flutter/engine/blob/cf0db3e54687e071e657771bc13acfbdc6af60a3/runtime/dart_snapshot.cc#L18 are needed and those should be global symbols

Reply:
thanks, yep these are global and would be left after `strip -x`:

```
nm -a ./build-strip-s-x/ios/Release-iphoneos/App.framework/App      

aarch64:
0000000000371fc0 R _kDartIsolateSnapshotData
000000000000a570 T _kDartIsolateSnapshotInstructions
000000000036a0e0 R _kDartVmSnapshotData
0000000000005400 T _kDartVmSnapshotInstructions
                 U dyld_stub_binder
```

I'll propose the change in another issue/PR.

Reply:
Created a follow up PR https://github.com/flutter/flutter/pull/111264